Source/Scene/modelMaterialsCommon.js

```diff
@@ -237,6 +237,11 @@ define([
             }
         }
 
+        // Give the diffuse uniform a semantic to support color replacement in 3D Tiles
+        if (defined(techniqueParameters.diffuse)) {
```

Comment:
It's not for backwards compatibility. The glTFs that go through here won't have techniques, so they won't have semantics already defined.